Kernel/Thread.h

```diff
@@ -316,38 +316,22 @@ template<typename Callback>
 inline IterationDecision Thread::for_each_in_state(State state, Callback callback)
 {
     ASSERT_INTERRUPTS_DISABLED();
-    for (auto* thread = thread_list_for_state(state)->head(); thread;) {
-        auto* next_thread = thread->next();
-        if (thread->state() == state) {
-            if (callback(*thread) == IterationDecision::Break)
-                return IterationDecision::Break;
-        }
-        thread = next_thread;
-    }
-
-    return IterationDecision::Continue;
+    if (is_runnable_state(state))
+        return for_each_runnable(callback);
+    else
+        return for_each_nonrunnable(callback);
 }
 
 template<typename Callback>
 inline IterationDecision Thread::for_each_living(Callback callback)
 {
     ASSERT_INTERRUPTS_DISABLED();
-    for (auto* thread = g_runnable_threads->head(); thread;) {
-        auto* next_thread = thread->next();
-        if (thread->state() != Thread::State::Dead && thread->state() != Thread::State::Dying)
-            if (callback(*thread) == IterationDecision::Break)
+    return Thread::for_each([callback](Thread& thread) -> IterationDecision {
+        if (thread.state() != Thread::State::Dead && thread.state() != Thread::State::Dying)
+            if (callback(thread) == IterationDecision::Break)
```

Comment:
When we nest multiple `if` like this, let's at the very least put `{` and `}` for the outer one.

Reply:
Noted, but since you pointed out I can just remove the if () altogether, I did that instead :)